### PR TITLE
Yunyang/testing/UI testing1

### PR DESCRIPTION
**Pull request by Yunyang-Lu**
First 3 UI tests for ViewListActivity is running, please check if its performance meets requirements.

### CODE REVIEW

File: code/app/src/androidTest/java/com/example/sigma_blue/ViewListActivityUITest.java [hunk 1]
```diff
@@ -38,9 +41,9 @@
...
 @LargeTest
 public class ViewListActivityUITest extends UITestTools {
-//    @Rule
-//    public ActivityScenarioRule<ViewListActivity> scenario = new
-//            ActivityScenarioRule<ViewListActivity>(ViewListActivity.class);
+    @Rule
+    public ActivityScenarioRule<LoginPageActivity> scenario = new
+            ActivityScenarioRule<LoginPageActivity>(LoginPageActivity.class);
 //
 //
```
Comment (waridh) [inline code/app/src/androidTest/java/com/example/sigma_blue/ViewListActivityUITest.java:48]: Hi, please remove the commented out code as well. I know they aren't yours.

File: code/app/src/androidTest/java/com/example/sigma_blue/ViewListActivityUITest.java [hunk 2]
```diff
@@ -119,79 +122,163 @@ public class ViewListActivityUITest extends UITestTools {
...
+        testUser_Login();
+        // get to edit page
+        onView(withId(R.id.addButton)).perform(click());
+        //onView(withId(R.id.button_edit)).perform(click());
+        // enter item info
+        onView(withId(R.id.text_name_disp)).perform(ViewActions.typeText("iName"));
+        onView(ViewMatchers.isRoot()).perform(closeSoftKeyboard());
+        onView(withId(R.id.text_value_disp)).perform(ViewActions.replaceText("100"));
+        onView(ViewMatchers.isRoot()).perform(closeSoftKeyboard());
+        onView(withId(R.id.text_make_disp)).perform(ViewActions.typeText("Banana"));
```
Comment (waridh) [inline code/app/src/androidTest/java/com/example/sigma_blue/ViewListActivityUITest.java:164]: Typing text into an edit text and then closing the keyboard gets repeated so much that it should probably be factored out into a method.

**Review by waridh — CHANGES_REQUESTED** [review]
Actual testing looks good. To make it easier for other people to potentially work on it, just making a the two changes that I suggested would really streamline the codebase.

File: code/app/src/androidTest/java/com/example/sigma_blue/ViewListActivityUITest.java (hunk 1 above)
Comment (waridh) [reply code/app/src/androidTest/java/com/example/sigma_blue/ViewListActivityUITest.java:48]: as in, I think I put them there, but we should have them removed 😅

**Comment by lizard-52** [comment]
Both tests run but don't pass. The account was missing, but even after adding a new testing account they don't seem to work properly.

**Comment by waridh** [comment]
> Both tests run but don't pass. The account was missing, but even after adding a new testing account they don't seem to work properly.

Do you think its because multiple people can't run the test at the same time?

**Comment by lizard-52** [comment]
> > Both tests run but don't pass. The account was missing, but even after adding a new testing account they don't seem to work properly.
> 
> Do you think its because multiple people can't run the test at the same time?

I was actually on an old commit, looks good now

**Comment by Yunyang-Lu** [comment]
> > Both tests run but don't pass. The account was missing, but even after adding a new testing account they don't seem to work properly.
> 
> Do you think its because multiple people can't run the test at the same time?

It can be an issue when there is an error occurs, cause the test account do not get deleted. After that any test will not pass
Current solution is login to firebase, and manually delete the account 
Which is bad, but I don't see any better solution can be done

**Comment by waridh** [comment]
> > > Both tests run but don't pass. The account was missing, but even after adding a new testing account they don't seem to work properly.
> > 
> > 
> > Do you think its because multiple people can't run the test at the same time?
> 
> It can be an issue when there is an error occurs, cause the test account do not get deleted. After that any test will not pass Current solution is login to firebase, and manually delete the account Which is bad, but I don't see any better solution can be done

That's fine for now. Automation on that would require us opening the backend of the app somehow. It sounds possible, but not in the due date.

**Comment by waridh** [comment]
> > > > Both tests run but don't pass. The account was missing, but even after adding a new testing account they don't seem to work properly.
> > > 
> > > 
> > > Do you think its because multiple people can't run the test at the same time?
> > 
> > 
> > It can be an issue when there is an error occurs, cause the test account do not get deleted. After that any test will not pass Current solution is login to firebase, and manually delete the account Which is bad, but I don't see any better solution can be done
> 
> That's fine for now. Automation on that would require us opening the backend of the app somehow. It sounds possible, but not in the due date.

Please clean up the code so that we could push this through. Clean code should be in our testing as well.

**Pull request merged**